### PR TITLE
Some more small improvements

### PR DESCRIPTION
**Pull request by zjeffer**

### CODE REVIEW

File: src/modules/hyprland/workspaces.cpp [hunk 1]
```diff
@@ -167,19 +167,18 @@ Workspaces::~Workspaces() {
...
-
-  if (name_.find("name:") == 0) {
+Workspace::Workspace(const Json::Value &workspace_data)
+    : id_(workspace_data["id"].asInt()),
+      name_(workspace_data["name"].asString()),
+      output_(workspace_data["monitor"].asString()),  // TODO:allow using monitor desc
+      windows_(workspace_data["id"].asInt()),
+      active_(true),
+      is_special_(false) {
+  if (name_.starts_with("name:")) {
```
Comment (zjeffer) [inline src/modules/hyprland/workspaces.cpp:177]: I had to compile with `-Dstd_cpp=c++20` for this to work, maybe remove this if you still want it to work on C++ 17.

File: src/modules/hyprland/workspaces.cpp [hunk 2]
```diff
@@ -167,19 +167,18 @@ Workspaces::~Workspaces() {
...
     name_ = name_.substr(5);
   } else if (name_.find("special") == 0) {
```
Comment (zjeffer) [inline src/modules/hyprland/workspaces.cpp:179]: And I just noticed I forgot to change the same thing here.

File: src/modules/hyprland/workspaces.cpp (hunk 1 above)
Comment (MightyPlaza) [reply src/modules/hyprland/workspaces.cpp:177]: maybe it's better to add/not add this after this PR has been resolved
https://github.com/Alexays/Waybar/pull/2322

Comment (zjeffer) [reply src/modules/hyprland/workspaces.cpp]: Alright, I removed it.

**Pull request merged**